### DISCUSSION TITLE
Only allow author to select an answer

### DISCUSSION THREAD

**Discussion by Billy-** [community/community#3190] [Discussions] [answered]
I have found it to be a frustration that anyone other than myself (as the author of a discussion) can mark a correct answer. Only I can truly say if someone's answer has answered my question/issue, surely?

Edit:
- Yes, only maintainers/admins are able to do this
- as mentioned in comments below, the OP can _unmark_ the answer as correct, which I did not realise. This solves the issue 🙂

**Comment by evi-liu** [comment, marked as answer]
Do you have examples of others mis-marking an answer? Currently, only the OP (post author) and those with maintainer/admin privileges may mark as answer.

**Reply by kovitikus** [reply]
I agree that it's really weird that a maintainer can come in and respond and then mark their own response as the answer.

I don't have a specific example to link you to, but there's many times when I've asked a question that was misunderstood and the answer wasn't what I needed to know. Miscommunication could be on the part of the person asking a question or the person answering, but sometimes it's also that the person asking doesn't entirely understand what they should be asking until they get some conversation going.

It's up to the person who posted the question to determine if a response has met their specifications for what constitutes an answer.

That being said, the author of the thread can unmark it as answered, if they disagree. A friend and I tested it out a bit. I don't know if this sort of marking battle is a good thing or a bad thing. It's certainly curious to say the least.

![Marking and unmarking an answered question](https://i.imgur.com/T43PKGn.png)

**Reply by Griatch** [reply]
On the contrary, I don't think it's strange that an admin can mark answered as well as the post author (if anyone *else* could, that'd be weird though). The obvious use case for the admin marking would be cleanup - e.g. the post author got an answer but  didn't bother to mark it for a long time. 

Admins marking their own answers is more of a community-culture thing (just decide that it should not be done lightly) - and since the post author can always edit a marking (also if set by an admin), it should be self-correcting. 

(the example in the image above was us testing the functionality btw)

**Reply by Billy-** [reply]
Yes it's the maintainer/admin privs that allow it. I did not realise you can unmark the answer! That pretty much resolves the issue.

**Reply by Griatch** [reply]
@Billy- So maybe it's the un-marking ability that needs to be clearer ... :wink:

**Reply by kovitikus** [reply]
I think both behaviors should be made clear, but until a person participates in a GitHub discussion I'm not certain how that's going to be possible. Baptism by fire I suppose.

@Griatch makes some very valid points and I agree that each community will have their own needs, so leaving it as is allows the most amount of flexibility. It will be up to each community to inform their users how a Q/A section functions to avoid confusion. A pinned "How to use Q/A." type thread should suffice to lay out the rules.
